### ISSUE TITLE
Adding regularization to the loss results in LLVM error when using CuArrays

### ISSUE THREAD

**Issue by salbert83** [FluxML/Flux.jl#787]
For mlp.jl from the model-zoo, https://github.com/FluxML/model-zoo/blob/master/vision/mnist/mlp.jl, I modified line 5 to:

using LinearAlgebra

and line 22 to 

loss(x,y) = crossentropy(m(x), y) + sum(norm, params(m))

The code runs without producing an error.  However, if I uncomment line 4, line 30 produces the error:

LLVM error: Cannot select: 0x35624040: f32 = fpow 0x35624fb0, 0x35624588

The error goes away if I remove the regularization term and just use the crossentropy.

CuArrays unit tests all pass on my machine, and I've verified that I'm able to do some GPU computations without issue.  So, why does Flux.train! fail?

**Comment by devmotion** [comment]
I'm not sure what's the underlying problem here, but you can circumvent the issue by defining
```julia
LinearAlgebra.norm(x::TrackedArray{T}) where T = sqrt(sum(abs2.(x)) + eps(T))
```
I guess it would make sense to add this special case to https://github.com/FluxML/Tracker.jl/pull/36. Maybe this also yields a more efficient implementation for the standard case `p = 2`, apart from solving the GPU problem.
